### PR TITLE
Story/io 240/calculate sums for planning view

### PR DESCRIPTION
**Pull request by EngbergJonas**
- calculate budgets for each project row
- calculate budgets for each class row
- calculate budgets for the PlanningSummaryTable (renamed from PlanningYearsTable)

### CODE REVIEW

File: src/utils/calculations.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+ * - key: the key for the cell (key of the finance object)
+ * - plannedBudget: sum of all underlying project finances for that year (calculated in the backend)
+ * - frameBudget: the amount of frameBudget given to the underlying projects during that year
+ * - deviation: the deviation between the plannedBudget and the framedBudget and if the value is negative
+ */
+export const calculatePlanningSummaryCells = (
+  classes: Array<IClass>,
+  type: PlanningRowType,
+): Array<IPlanningCell> => {
+  const totalFinances = classes.reduce((acc: IClassFinances, curr: IClass) => {
```
Comment (kgunga) [inline src/utils/calculations.ts:101]: Rename acc and curr arguments to appropriate names

**Review by kgunga — CHANGES_REQUESTED** [review]

**Review by kgunga — CHANGES_REQUESTED** [review]

File: src/components/Planning/PlanningTable/PlanningRow/ProjectRow/ProjectCell.tsx [hunk 2]
```diff
@@ -286,29 +284,38 @@ const ProjectCell: FC<IProjectCellProps> = ({ cell, onUpdateProject }) => {
...
 
-  // Make sure the budget is always at least 0 and remove the first 0 if the user types a number in the field
-  const handleChange = useCallback(
-    (e: ChangeEvent<HTMLInputElement>) => setFormValue(e.target.value ? +e.target.value : 0),
-    [],
-  );
+  // Removes the zero value on change if there is only one zero in the value
+  const handleChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+    // If value is greater than 1 and starts with a zero, let the value be
+    if (/^0.{2,}$/.exec(e.target.value)) {
```
[the remark this answers is not in the document]
Comment (EngbergJonas) [reply src/components/Planning/PlanningTable/PlanningRow/ProjectRow/ProjectCell.tsx:306]: changed comment to "longer"

File: src/components/Planning/PlanningSummaryTable/PlanningSummaryTable.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+              </div>
+            </td>
+          ))}
+        </tr>
+      </tbody>
+    </table>
+  );
+};
+
+export default PlanningSummaryTable;
```
[the remark this answers is not in the document]
Comment (EngbergJonas) [reply src/components/Planning/PlanningSummaryTable/PlanningSummaryTable.tsx]: fixed

File: src/utils/calculations.ts (hunk 1 above)
Comment (EngbergJonas) [reply src/utils/calculations.ts]: fixed

File: src/components/Planning/PlanningInfoPanel/PlanningInfoPanel.tsx [hunk 4]
```diff
@@ -18,35 +18,37 @@ const PlanningInfoPanel: FC<IPlanningInfoPanelProps> = ({ selectedMasterClass })
...
+            {t('planning')}
+          </Button>
+        </div>
+        <div id="previousButton" data-testid="previous-button-container">
+          {selectedMasterClass && (
+            <Button
+              data-testid="previous-button"
+              onClick={navigateBack}
+              variant="supplementary"
+              iconLeft={<IconAngleLeft />}
```
[the remark this answers is not in the document]
Comment (EngbergJonas) [reply src/components/Planning/PlanningInfoPanel/PlanningInfoPanel.tsx]: fixed

**Review by kgunga — APPROVED** [review]
👍

**Pull request merged**